### PR TITLE
Convert boost::filesystem references to std::filesystem

### PR DESCRIPTION
**Pull request by meyediri**
associated issue #4057

### CODE REVIEW

File: nano/nano_rpc/CMakeLists.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@ target_link_libraries(
   nano_rpc
   rpc
   secure
-  Boost::filesystem
+  std::filesystem
```
Comment (clemahieu) [inline nano/nano_rpc/CMakeLists.txt:7]: This line should be removed.

File: nano/core_test/common.hpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#pragma once
```
Comment (clemahieu) [inline nano/core_test/common.hpp:1]: It looks like some other files got mixed in.

**Review by clemahieu — CHANGES_REQUESTED** [review]
Also needs a formatting run ci/clang-format-do.sh

**Comment by qwahzi** [comment]
Closed by #4057 and #4315

**Pull request closed without merging**